src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratorExecutionVersion.cs

```diff
@@ -103,6 +103,19 @@ public static SourceGeneratorExecutionVersionMap Deserialize(ObjectReader reader
         return new(builder.ToImmutable());
     }
 
+    public Checksum GetChecksum()
+    {
+        using var _ = ArrayBuilder<Checksum>.GetInstance(this.Map.Count * 2, out var checksums);
+
+        foreach (var (projectId, version) in this.Map)
+        {
+            checksums.Add(projectId.Checksum);
+            checksums.Add(Checksum.Create(version, static (v, w) => v.WriteTo(w)));
+        }
+
+        return Checksum.Create(checksums);
```

Comment:
>Checksum.Create(checksums); [](http://example.com/codeflow?start=15&length=27)

a bit off topic: is the ReadOnlySpan overload of Checksum.Create more performant than the ImmutableArray/ArrayBuilder versions? If so, and not that it helps here, couldn't the ImmutableArray overload get the underlying array and use the equivalent of the readonlyspan implementation?

Reply:
yes.  the IA version should defer to the ROS version.

That said, this is not an IA.  This is an ArrayBuilder.